presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -730,6 +733,40 @@ public PlanWithProperties visitJoin(JoinNode node, StreamPreferredProperties par
                 buildPreference = singleStream();
             }
             PlanWithProperties build = planAndEnforce(node.getRight(), buildPreference, buildPreference);
+            
+            if (preferMergeJoin(session)
```

Comment:
It seems like this change should belong in a new optimizer rule (similar to `IndexJoinOptimizer`)?  I.e. I'm not sure how this relates to adding local exchange?

Reply:
So basically we want to eliminate adding local exchange on top of TableScan for merge join

If we have hash based join, the plan would be
JoinNode
- TableScanA
- LocalExchange
   - TableScanB

But if the table property layout can meet merge join requirements, we don't need to add the LocalExchange, it would be
JoinNode (merge join)
- TableScanA
- TableScanB

Reply:
My point is the introduction of JoinNode (merge join) seems to be more appropriately done in a separate optimizer rule, with this one simply removing the local exchange as you say.

Reply:
Just move it to a new Optimizer rule, can you help review?